src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+    Xc = [denominator(f) for f in X[:, c]]
+    t = lcm(Xc)
+    result = multiply_column!(X, t, c)
+    return result
+
+end
+
+
+
+function matrix_clear_den(X::Oscar.MatElem)
+    rs, cs = size(X)
+    for c in 1:cs
+        X = matrix_clear_den_in_col(X, c)
+    end
+    return X
+end
+
+
+function reduce_ideal_one_step(MRS::MatroidRealizationSpace, 
+                               elim::Vector{<:RingElem}, 
+                               fullyReduced::Bool)
+
+    Igens = gens(MRS.defining_ideal)
+    Sgens = MRS.inequations
+    R = MRS.ambient_ring
+    FR = FractionField(R)
+    xs = gens(R)
+    X = MRS.realization_matrix
+    nr, nc = size(X)
+    
+    Ivars = ideal_vars(Igens);
+
+    for x in Ivars 
+        t = find_solution_v(x, Igens, Sgens, R)
+        t isa String && continue
+        
+        phi = sub_map(x, t, R, xs)
+        
+	    Sgens_new = n_new_Sgens(x, t, Sgens, R, xs);
+        if length(Sgens_new) == 0
+            Sgens_new = Vector{RingElem}()
+        end
+        Igens_new = n_new_Igens(x, t, Igens, Sgens_new, R, xs);
+        push!(elim, x)
+        
+        phiX = matrix(FR, [phi(X[i,j]) for i in 1:nr, j in 1:nc  ] )
+        nX_FR = matrix_clear_den(phiX)
+        nX = matrix(R, [numerator(nX_FR[i,j])  for i in 1:nr, j in 1:nc ])
+        
+        GBnew = collect(groebner_basis(ideal(R, Igens_new)))         
+        
+        MRS_new = MatroidRealizationSpace(ideal(R, GBnew), Sgens_new, R, nX, MRS.F, MRS.char, MRS.q )
+
+        
+        return (MRS_new, elim, fullyReduced)
+    end
+
+    return (MRS, elim, true)
+
+end
+
+
+function reduce_realization_space(MRS::MatroidRealizationSpace,
+                               elim::Vector{RingElem} = Vector{RingElem}(), 
+                               fullyReduced::Bool = false) 
+    
+    
+    #If there are no variables left, we don't reduce anything
+    if !(typeof(MRS.ambient_ring) <: MPolyRing)
+        return MRS
+    end
+
+    output = reduce_ideal_one_step(MRS, elim, fullyReduced)
+    output isa String && return "Not Realizable 0 in Semigroup"
+    (MRS, elim, fullyReduced) = output    
+    
+    !fullyReduced && return reduce_realization_space(MRS, elim, fullyReduced)
+    
+    
+    R = MRS.ambient_ring
+    xs = gens(R)
+    cR = coefficient_ring(R)
+    X = MRS.realization_matrix
+    nr, nc = size(X)
+    Igens = gens(MRS.defining_ideal)
+    Sgens = MRS.inequations
+
+    
+    zero_elim = []        
+    for i in 1:length(xs)
+        if xs[i] in elim
+            push!(zero_elim, 0)
+        else
+            push!(zero_elim, "x"*string(i) ) 
+        end
+    end
+        
+    xnew_str = Vector{String}(filter(x -> x!=0,  zero_elim))    
+    
+    
+    
+    if length(xnew_str) == 0
+        phi = hom(R, cR, a->a, [cR(0) for i in 1:length(xs)])
+        ambR = codomain(phi);
+        if length(Igens) == 0
+            Inew = ideal(ambR, ambR(0))
+        else
+            Inew = ideal(ambR, phi.(Igens)); 
+        end
```

Comment:
Huh, doesn't this work? If not then that seems like a bug to me.
```suggestion
        Inew = ideal(ambR, phi.(Igens))
```
Same comment applies to other parts of your code, too.

Reply:
The problem is that `phi.(Vector{RingElem}())` doesn't preserve the type and returns `Any[]`. Is that supposed to be the case. The list `Any[]` breaks subsequent parts of our code which always expect to receive lists of ring elements.

Reply:
What's the type of `phi`? This should become type-stable so that the result not `Any[]`. In either case, you could also do `Inew = ideal(ambR, ambR.(phi.(Igens)))` and avoid the `if`.